ios/keyman/Keyman/Keyman/MainViewController.swift

```diff
@@ -0,0 +1,1200 @@
+    textSizeTitle.text = "Text size: \(Int(textSize))"
+    textSizeTitle.textAlignment = .center
+    textSizeTitle.textColor = UIColor.gray
+    textSizeTitle.font = textSizeTitle.font.withSize(14.0)
+    containerView.addSubview(textSizeTitle)
+
+    let sliderFrame = CGRect(x: 12, y: 45, width: bframe.size.width - 24, height: 50)
+    textSizeController.frame = sliderFrame
+    containerView.addSubview(textSizeController)
+
+    textSizeVC.view.addSubview(containerView)
+    if UIDevice.current.userInterfaceIdiom == .phone {
+      wasKeyboardVisible = textView.isFirstResponder
+      textView.dismissKeyboard()
+      present(textSizeVC, animated: false, completion: {() -> Void in
+        UIView.animate(withDuration: 0.25, delay: 0.0, options: .curveEaseIn, animations: {() -> Void in
+          containerView.frame = eframe
+          textSizeVC.view.backgroundColor = UIColor(red: 0.0, green: 0.0, blue: 0.0, alpha: 0.4)
+        }, completion: nil)
+      })
+    } else {
+      let vc = UIViewController()
+      vc.modalPresentationStyle = .popover
+      vc.popoverPresentationController?.barButtonItem = navigationItem.rightBarButtonItems?[6]
+      vc.view = containerView
+      vc.preferredContentSize = containerView.frame.size
+      present(vc, animated: true, completion: nil)
+    }
+  }
+
+  func dismissTextSizeVC(_ sender: Any) {
+    let presentedVC = presentedViewController
+    if UIDevice.current.userInterfaceIdiom == .phone {
+      let containerView: UIView! = presentedVC?.view?.viewWithTag(1)
+      let sframe = containerView.frame
+      let eframe = CGRect(x: sframe.origin.x, y: sframe.origin.y + sframe.height + 8,
+                          width: sframe.width, height: sframe.height)
+      UIView.animate(withDuration: 0.25, delay: 0.0, options: .curveEaseIn, animations: {() -> Void in
+        containerView.frame = eframe
+        presentedVC?.view?.backgroundColor = UIColor.clear
+      }, completion: {(_ finished: Bool) -> Void in
+        presentedVC?.dismiss(animated: false, completion: {() -> Void in
+          if self.wasKeyboardVisible {
+            self.textView.becomeFirstResponder()
+          }
+        })
+      })
+    } else {
+      presentedVC?.dismiss(animated: true) { _ in }
+    }
+  }
+
+  func trashButtonClick(_ sender: Any) {
+    _ = dismissDropDownMenu()
+    let alert = UIAlertController(title: nil, message: nil, preferredStyle: .actionSheet)
+    let clear = UIAlertAction(title: "Clear Text", style: .destructive, handler: {(_ action: UIAlertAction) -> Void in
+      self.textView.text = ""
+    })
+    let cancel = UIAlertAction(title: "Cancel", style: .default, handler: nil)
+    alert.addAction(clear)
+    alert.addAction(cancel)
+    alert.popoverPresentationController?.barButtonItem = navigationItem.rightBarButtonItems?[4]
+    present(alert, animated: true, completion: nil)
+  }
+
+  var overlayWindow: UIWindow! {
+    return appDelegate?.overlayWindow
+  }
+
```

Comment:
A few methods related to 'actionSheet' stuff disappeared here.  If we are able to safely remove that part of the code, it's probably fine, but I'm not clear on what removing it changes.